File: smart-campus-backend/models/course.go
package models

import "gorm.io/gorm"

type Course struct {
	gorm.Model
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Lecturer    string    `json:"lecturer"`
	Students    []Student `json:"students" gorm:"many2many:student_courses;"`
}

File: mobile_app/lib/models/course.dart
class Course {
  final int id;
  final String name;
  final String description;
  final String lecturer;

  Course({required this.id, required this.name, required this.description, required this.lecturer});

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['ID'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      lecturer: json['lecturer'] ?? 'N/A',
    );
  }
}